### PR TITLE
Add a new load generation mode that produces DEX transactions.

### PR DESCRIPTION
**Pull request by dmkozh**
# Description

This allows to test setups that involve the mix of non-DEX and DEX traffic and its interaction with the DEX-limiting config setting.

Also refactored the loadgen code a bit and fixed the existing test.

# Checklist
- [ ] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [ ] Rebased on top of master (no merge commits)
- [ ] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [ ] Compiles
- [ ] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: src/simulation/LoadGenerator.cpp [hunk 1]
```diff
@@ -258,21 +262,69 @@ LoadGenerator::generateLoad(GeneratedLoadConfig cfg)
...
+                generateTx;
+
+            switch (cfg.mode)
+            {
+            case LoadGenMode::CREATE:
+                releaseAssert(false);
+                break;
+            case LoadGenMode::PAY:
+                generateTx = [&]() {
+                    return paymentTransaction(cfg.nAccounts, cfg.offset,
```
Comment (MonsieurNicolas) [inline src/simulation/LoadGenerator.cpp:286]: I think asan complains that you're capturing `sourceAccountId` and `opCount` but those variables go out of scope

Comment (dmkozh) [reply src/simulation/LoadGenerator.cpp:286]: Nice, your change has already saved me a couple of hours debugging time!

File: src/simulation/LoadGenerator.cpp [hunk 2]
```diff
@@ -382,23 +425,17 @@ LoadGenerator::submitPaymentOrPretendTx(GeneratedLoadConfig const& cfg,
             status != TransactionQueue::AddResult::ADD_STATUS_ERROR)
         {
             mFailed = true;
-            return 0;
+            return false;
```
Comment (MonsieurNicolas) [inline src/simulation/LoadGenerator.cpp:428]: you missed the `return cfg.nTxs;` above that should be `false` I think given the refactor (C++ is nice like that)

File: src/simulation/LoadGenerator.cpp [hunk 3]
```diff
@@ -258,21 +262,69 @@ LoadGenerator::generateLoad(GeneratedLoadConfig cfg)
...
+                    }
+                };
+            }
             break;
-        }
+            }
 
+            if (submitTx(cfg, generateTx))
+            {
+                --cfg.nTxs;
```
Comment (MonsieurNicolas) [inline src/simulation/LoadGenerator.cpp:325]: note that with this refactor, unlike the previous version, loadgen attempts to keep submitting transactions even after running into an error. I think this is fine @marta-lokhova can confirm

**Review by MonsieurNicolas — CHANGES_REQUESTED** [review]

File: src/main/CommandHandler.cpp [hunk 4]
```diff
@@ -989,7 +989,9 @@ CommandHandler::generateLoad(std::string const& params, std::string& retStr)
             parseOptionalParam<uint32_t>(map, "maxfeerate");
         cfg.skipLowFeeTxs =
             parseOptionalParamOrDefault<bool>(map, "skiplowfeetxs", false);
-
+        // Only for MIXED_TX mode; fraction of DEX transactions.
```
[the remark this answers is not in the document]
Comment (dmkozh) [reply src/main/CommandHandler.cpp:992]: Done.

File: src/simulation/LoadGenerator.cpp [hunk 5]
```diff
@@ -81,6 +81,10 @@ LoadGenerator::getMode(std::string const& mode)
     {
         return LoadGenMode::PRETEND;
     }
+    else if (mode == "mixed_txs")
```
[the remark this answers is not in the document]
Comment (dmkozh) [reply src/simulation/LoadGenerator.cpp:84]: Done.

File: src/simulation/LoadGenerator.cpp (hunk 2 above)
Comment (dmkozh) [reply src/simulation/LoadGenerator.cpp:428]: Nice catch, fixed.

**Review by MonsieurNicolas — APPROVED** [review]
LGTM.
Please rebase & squash so that we can merge

File: src/simulation/LoadGenerator.cpp (hunk 3 above)
Comment (marta-lokhova) [reply src/simulation/LoadGenerator.cpp:325]: Sorry I didn't see this earlier. I don't think we should keep submitting transactions after failures. We have some missions in CI that wait until loadgen succeeds or fails so the mission can "reset" core and retry. The expectation is that after completion no new tx load is generated by loadgen.

Comment (dmkozh) [reply src/simulation/LoadGenerator.cpp:325]: Actually I didn't intend to change the behavior here; it was just a bug. Added a break on failure.

**Comment by dmkozh** [comment]
> LGTM. Please rebase & squash so that we can merge

Done.

**Comment by MonsieurNicolas** [comment]
r+ adaed55a088a7ab70083cc5483074f66d35700bb

**Pull request merged**